test/Microsoft.NET.Build.Tests/GivenThatWeWantToDetectDeprecatedRoslynCompilerType.cs

```diff
@@ -0,0 +1,171 @@
+                IsExe = true
+            };
+
+            testProject.AdditionalProperties["RoslynCompilerType"] = "Core";
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testAsset);
+
+            var result = buildCommand
+                .Execute();
+
+            result
+                .Should()
+                .Pass()
+                .And
+                .NotHaveStdOutContaining("NETSDK1234");
+        }
+
+        [Fact]
+        public void It_does_not_warn_when_RoslynCompilerType_is_FrameworkPackage()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "RoslynCompilerTypeFrameworkPackage",
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework,
+                IsExe = true
+            };
+
+            testProject.AdditionalProperties["RoslynCompilerType"] = "FrameworkPackage";
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testAsset);
+
+            var result = buildCommand
+                .Execute();
+
+            result
+                .Should()
+                .Pass()
+                .And
+                .NotHaveStdOutContaining("NETSDK1234");
+        }
+
+        [Fact]
+        public void It_does_not_warn_when_RoslynCompilerType_is_not_set()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "RoslynCompilerTypeNotSet",
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework,
+                IsExe = true
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testAsset);
+
+            var result = buildCommand
+                .Execute();
+
+            result
+                .Should()
+                .Pass()
+                .And
+                .NotHaveStdOutContaining("NETSDK1234");
+        }
+
+        [Fact]
+        public void It_does_not_warn_when_suppressed_with_NoWarn()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "RoslynCompilerTypeNoWarn",
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework,
+                IsExe = true
+            };
+
+            testProject.AdditionalProperties["RoslynCompilerType"] = "Framework";
+            testProject.AdditionalProperties["NoWarn"] = "NETSDK1234";
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testAsset);
+
+            var result = buildCommand
+                .Execute();
+
+            result
+                .Should()
+                .Pass()
+                .And
+                .NotHaveStdOutContaining("NETSDK1234");
+        }
+
+        [Fact]
+        public void It_can_suppress_warning_via_command_line()
```

Comment:
Looks like both suppression tests are failing. I think NETSDK warnings must be suppressed via `MSBuildWarningsAsMessages`